config/appstructure.go

```diff
@@ -0,0 +1,929 @@
+	if !strings.HasSuffix(listObjectPath, "/qListObjectDef") {
+		listObjectPath = fmt.Sprintf("%s/qListObjectDef", def.DataDef.Path)
+	}
+	listObjects := senseobjdef.NewDataPath(listObjectPath)
+
+	// Try to set dimensions and measures, null if not exist or not parsable (error)
+	rawDimensions, _ := dimensions.Lookup(properties)
+	if rawDimensions != nil {
+		var dimensions []*enigma.NxDimension
+		if err := jsonit.Unmarshal(rawDimensions, &dimensions); err != nil {
+			return errors.WithStack(err)
+		}
+		obj.Dimensions = make([]AppStructureDimensionMeta, 0, len(dimensions))
+		for _, dimension := range dimensions {
+			if dimension == nil {
+				continue
+			}
+
+			obj.Dimensions = append(obj.Dimensions, AppStructureDimensionMeta{
+				LibraryId:       dimension.LibraryId,
+				LabelExpression: dimension.Def.LabelExpression,
+				Defs:            dimension.Def.FieldDefs,
+				Labels:          dimension.Def.FieldLabels,
+			})
+		}
+	}
+
+	rawMeasures, _ := measures.Lookup(properties)
+	if rawMeasures != nil {
+		var measures []*enigma.NxMeasure
+		if err := jsonit.Unmarshal(rawMeasures, &measures); err != nil {
+			return errors.WithStack(err)
+		}
+		obj.Measures = make([]AppStructureMeasureMeta, 0, len(measures))
+		for _, measure := range measures {
+			if measure == nil {
+				continue
+			}
+			obj.Measures = append(obj.Measures, AppStructureMeasureMeta{
+				LibraryId: measure.LibraryId,
+				Label:     measure.Def.Label,
+				Def:       measure.Def.Def,
+			})
+		}
+	}
+
+	rawListObject, _ := listObjects.Lookup(properties)
+	if rawListObject != nil {
+		var listObject enigma.ListObjectDef
+		if err := jsonit.Unmarshal(rawListObject, &listObject); err != nil {
+			return errors.WithStack(err)
+		}
+		obj.Dimensions = []AppStructureDimensionMeta{
+			{
+				LibraryId:       listObject.LibraryId,
+				LabelExpression: listObject.Def.LabelExpression,
+				Defs:            listObject.Def.FieldDefs,
+				Labels:          listObject.Def.FieldLabels,
+			},
+		}
+
+		if obj.Measures == nil {
+			obj.Measures = make([]AppStructureMeasureMeta, 0, len(listObject.Expressions))
+		}
+
+		for _, expression := range listObject.Expressions {
+			obj.Measures = append(obj.Measures, AppStructureMeasureMeta{
+				LibraryId: expression.LibraryId,
+				Def:       expression.Expr,
+			})
+		}
+	}
+
+	if obj.Selectable && (len(obj.Dimensions)+len(obj.Measures)) < 1 {
+		// object defined as selectable both doesn't have any data definitions found
+		structure.warn(fmt.Sprintf("object<%s> visualization<%s> type<%s> is expected to have data, but no measures or dimensions were found", obj.Id, vis, typ))
+	}
+
+	// no dimension = not selectable
+	if len(obj.Dimensions) < 1 {
+		obj.Selectable = false
+	}
+
+	return nil
+}
+
+func (structure *AppStructure) handleMeasure(ctx context.Context, app *senseobjects.App, id string, obj *AppStructureObject) error {
+	genMeasure, err := app.Doc.GetMeasure(ctx, id)
+	if err != nil {
+		fmt.Printf("Measure: %+v\n", genMeasure)
```

Comment:
Not logging?

Reply:
oops.. it's logged through error returned, was left in there for debugging, removing it.